src/algorithms/local_search/local_search.cpp

```diff
@@ -1330,6 +1387,11 @@ void LocalSearch<Route,
                           return sum + _sol_state.route_costs[c];
                         });
       assert(new_cost + best_gain == previous_cost);
+      
+      for (auto v_rank : update_candidates) {
+        assert(_sol[v_rank].size() <= _input.vehicles[v_rank].max_number_of_tasks);
```

Comment:
Useful addition! We could avoid this loop altogether by moving the assert to the similar loop right above (line 1379).